File: Flutter MARN-Project-main/graduation_project/lib/features/chatboot/data/data_sources/assistant_service.dart
import 'package:MARN/features/chatboot/data/models/assistant_message_model.dart';
import 'package:MARN/features/chatboot/data/models/assistant_session_model.dart';

class SendMessageResponse {
  final String sessionId;
  final AssistantMessageModel assistantMessage;

  SendMessageResponse({
    required this.sessionId,
    required this.assistantMessage,
  });
}

abstract class AssistantService {
  Future<SendMessageResponse> sendMessage({
    required String? sessionId,
    required String content,
  });

  Future<List<AssistantSessionModel>> getSessions();

  Future<List<AssistantMessageModel>> getSessionMessages({
    required String sessionId,
  });

  Future<AssistantSessionModel> renameSession({
    required String sessionId,
    required String sessionName,
  });
}

File: MARN-React-main/src/services/assistantService.ts
import { apiClient } from './apiClient'
import type { ApiResponse } from '@/types/common'
import type { components } from '@/types/api.generated'

export interface AssistantSession {
  sessionId: string
  sessionName?: string | null
  createdAt?: string
}

export interface AssistantMessage {
  role: 'user' | 'assistant'
  content: string
  createdAt?: string
}

export interface SendMessageResult {
  sessionId: string
}

type SendMessagePayload = components['schemas']['SendAssistantMessageRequestDto']
type RenameSessionPayload = components['schemas']['RenameAssistantSessionRequestDto']

export const assistantService = {
  getSessions: () =>
    apiClient.get<ApiResponse<AssistantSession[]>>('/api/Assistant/sessions'),

  sendMessage: (payload: SendMessagePayload) =>
    apiClient.post<ApiResponse<SendMessageResult>>('/api/Assistant/messages', payload),

  getSessionMessages: (sessionId: string) =>
    apiClient.get<ApiResponse<AssistantMessage[]>>(
      `/api/Assistant/sessions/${sessionId}/messages`,
    ),

  renameSession: (sessionId: string, payload: RenameSessionPayload) =>
    apiClient.patch(`/api/Assistant/sessions/${sessionId}/name`, payload),
}
